### PR TITLE
Tags and services

### PR DESCRIPTION
**Pull request by aljesusg**

### CODE REVIEW

File: app/models/manageiq/providers/consumption/consumption_manager.rb [hunk 1]
```diff
@@ -64,6 +64,9 @@ def self.generate_event_resource(resource, date)
         }
     )
     e.save!
```
Comment (sergio-ocon) [inline app/models/manageiq/providers/consumption/consumption_manager.rb]: Shouldn't you save the object after modifying it, and not before?

File: app/models/manageiq/providers/consumption/consumption_manager/showback_event.rb [hunk 2]
```diff
@@ -34,14 +34,30 @@ def get_measure(category, dimension)
 
   def update_event
     generate_data unless self.data.present?
-    @metrics = metrics_time_range(end_time,start_time.end_of_month)
-    self.data.each do |key,dimensions|
-      dimensions.keys.each do |dim|
-        self.data[key][dim] = self.send("#{key}_#{dim}", data[key][dim].to_d)
+    if resource.methods.include?(:metrics)
```
Comment (sergio-ocon) [inline app/models/manageiq/providers/consumption/consumption_manager/showback_event.rb]: if resource.methods does not include :metrics? Shouldn't we look at the envelope? How do you do non-CU data?

File: app/models/manageiq/providers/consumption/consumption_manager/showback_event.rb [hunk 3]
```diff
@@ -90,10 +106,10 @@ def get_parent(res, look)
   HARDWARE_RESOURCE = %w(Vm Host EmsCluster ExtManagementSystem ).freeze
   CONTAINER_RESOURCE = %w(Container ContainerNode ContainerProject ExtManagementSystem).freeze
 
-  def assign_resource(resource)
-    bucket = find_bucket(resource)
-    bucket&.add_event(self)
-    resource_type = resource.type.split("::")[-1]
+  def assign_resource
+    bucket = find_bucket(resource)&.add_event(self)
+    resource_type = nil
```
Comment (sergio-ocon) [inline app/models/manageiq/providers/consumption/consumption_manager/showback_event.rb]: What is this for?

File: app/models/manageiq/providers/consumption/consumption_manager/showback_event.rb (hunk 2 above)
Comment (aljesusg) [inline app/models/manageiq/providers/consumption/consumption_manager/showback_event.rb]: Service has no metrics.

**Comment by sergio-ocon** [comment]
:)

**Pull request merged**